autoendpoint/src/server/routes/registration.rs

```diff
@@ -108,6 +109,38 @@ pub async fn update_token_route(
     Ok(HttpResponse::Ok().finish())
 }
 
+/// Handle the `POST /v1/{router_type}/{app_id}/registration/{uaid}/subscription` route
+pub async fn new_channel_route(
+    _auth: AuthorizationCheck,
+    path_args: RegistrationPathArgsWithUaid,
+    channel_data: Option<Json<NewChannelData>>,
+    state: Data<ServerState>,
+) -> ApiResult<HttpResponse> {
+    // Add the channel
+    debug!("Adding a channel to UAID {}", path_args.uaid);
+    let channel_data = channel_data.map(Json::into_inner).unwrap_or_default();
+    let channel_id = channel_data.channel_id.unwrap_or_else(Uuid::new_v4);
```

Comment:
It's really important that we don't modify how we get the channelid. The client stores and handles this as a string, and if we change the format, the matches fail. 

This does mean that we should probably just store the channelid as a string with appropriate comments.

Reply:
Can we fix the client so it parses the UUID as a UUID? Looking at the Python FCM code, it always sends the channel ID in the message as a simple hex string, so I'm confused where these "matches" you mention live?

Reply:
I think these matches are in Sync. When a sync client does a PUT it can choose the format of the uuid (unfortunately). We could normalize the uuid representation in the backend, but then when we go to generate urls we can't generate the uuid format the client is expecting.

This is only a problem because sync is an open protocol and people write their own clients.

I'm not sure if this same problem exists in push. I don't think push has to generate urls with uuids in them to give back to the client, but I'm not positive.

Reply:
This is fixed in the latest commit (return lowercase hyphenated UUIDs to match the client's expected format). Additionally, the UUIDs returned during user registration (#185) were adjusted.